pandas/core/groupby/generic.py

```diff
@@ -1754,10 +1754,22 @@ def value_counts(
                     level=index_level, sort_remaining=False
                 )
 
-            if not self.as_index:
+            if self.as_index:
+                return result.__finalize__(self.obj, method="value_counts")
+            else:
                 # Convert to frame
-                result = result.reset_index(name="proportion" if normalize else "count")
-            return result.__finalize__(self.obj, method="value_counts")
+                name = "proportion" if normalize else "count"
+                index = result.index
+                columns = com.fill_missing_names(index.names)
+                if name in columns:
+                    raise ValueError(
+                        f"Column label '{name}' is duplicate of result column"
+                    )
+                result.name = name
+                result.index = index.set_names(range(len(columns)))
```

Comment:
again this seems way more complicated here. can you not simply
```result_frame.columns = ....```

Reply:
Done